src/erasure-code/ErasureCodePlugin.cc

```diff
@@ -16,22 +16,18 @@
  */
 
 #include <errno.h>
-#include <dlfcn.h>
 
 #include "ceph_ver.h"
 #include "ErasureCodePlugin.h"
 #include "common/errno.h"
+#include "common/shared_lib.h"
```

Comment:
we could have a `dlfcn_compat.h` which includes `<dlfcn.h>` on platforms which supports dlfcn functions, and include the header which introduces the homebrew shims otherwise.

Reply:
That IMO can become even more confusing, as we'd be including ``dlfcn_compat.h`` but use ``shared_lib_*`` functions.

Reply:
sorry for the confusion, i assume that we go with the drop-in-replacement approach.

Reply:
Oh, right. Makes sense then.